api/src/domain/license-status/searchLicenseStatusFactory.ts

```diff
@@ -1,84 +1,18 @@
-import { getLicenseDate, parseDateWithFormat } from "@shared/dateHelpers";
-import { LicenseStatus, LicenseStatusItem, LicenseStatusResult, NameAndAddress } from "@shared/license";
-import { inputManipulator } from "../inputManipulator";
-import { LicenseStatusClient, SearchLicenseStatus } from "../types";
-
-export const searchLicenseStatusFactory = (licenseStatusClient: LicenseStatusClient): SearchLicenseStatus => {
-  return async (nameAndAddress: NameAndAddress, licenseType: string): Promise<LicenseStatusResult> => {
-    const searchName = inputManipulator(nameAndAddress.name)
-      .makeLowerCase()
-      .removeBusinessDesignators()
-      .trimPunctuation().value;
-
-    const entities = await licenseStatusClient.search(searchName, nameAndAddress.zipCode, licenseType);
-
-    const allMatchingAddressesArray = entities.filter((it) => {
-      return cleanAddress(it.addressLine1).startsWith(cleanAddress(nameAndAddress.addressLine1));
-    });
-
-    if (allMatchingAddressesArray.length === 0) {
-      throw new Error("NO_MATCH");
+import { SearchLicenseStatus } from "../types";
+
+export const searchLicenseStatusFactory = (
+  webServiceSearchLicenseStatus: SearchLicenseStatus,
+  dynamicsSearchLicenseStatus: SearchLicenseStatus
```

Comment:
[sand] Small & picky but for pure DI, I'd really want to avoid referencing any specific implementations by name anywhere in our domain code. Like, this approach is injecting them properly (that's DI) but just the variable naming is still making this code "verbally depend" on the concept of dynamics specifically.

We could call them ("legacy" and "new") or even ("1" and "2"). Idk. This is probably the one place in the whole code you'll ever hear me arguing AGAINST more specific, explicit, readable variable naming, but I feel like it's not right to go to these lengths to have our domain code decoupled from specific implementations, and then have the variable name undo that by referencing the specific implementations directly.

Reply:
I understand the decoupling of the name and I thought about that as I was building this as well for the same reason. I was hesitant as the test names felt not specific enough? Do we change it so it just says`returns new license status client when env var is true`, is that okay?